docs/core-concepts/route-params.md

```diff
@@ -136,27 +134,88 @@ const idFormatParam: ParamGetSet<IdFormat> = {
 If you define your path params with a question mark, `:?` the router assumes this param is not required.
 
 ```ts
-const routes = [
+const routes = createRoutes([
   {
     name: 'users',
     path: '/users/:?id', // [!code focus]
     component: ...
   }
-] as const satisfies Routes
+])
 ```
 
 This means the route will match even if nothing is provided after `/users/`.
 
 This also works for non-string params.
 
 ```ts
-const routes = [
+const routes = createRoutes([
   {
     name: 'users',
     path: path('/users/:?id', { id: Number }), // [!code focus]
     component: ...
   }
-] as const satisfies Routes
+])
 ```
 
 Which now, `router.params.id` has the type `number | undefined`, and will only match url where the value passes as a number or is missing entirely.
+
+## Param Name
+
+There are very few constraints on the name you choose for a param. Your param name can include any special character except forward slash `/`. Kitbag Router uses forward slash (or the end of the string) to denote the end of a param name in a path.
+
+```ts
+const routes = createRoutes([
+  {
+    name: 'users',
+    path: '/users/:can-have#what.ever!?',
+    component: ...
+  }
+])
+```
+
+Keep in mind that any special characters in the param name will make accessing the value slightly less pretty.
+
+```ts
+const route = useRoute()
+
+route.params['can-have#what.ever!?']
+```
+
+If your param name starts with a question mark `?`, Router will assume your param was intended to be optional.
```

Comment:
I would reword this to just state that prefixing your param name with a ? will make it optional.

Reply:
I tried 🤷 see what you think now